### PR TITLE
Web API: ignore invalid timestamp value /w schema

### PR DESCRIPTION
**Pull request by de-code**
related to https://github.com/elifesciences/data-hub-issues/issues/577

### CODE REVIEW

File: tests/unit_test/generic_web_api/transform_data_test.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                    "name": "ts",
+                    "type": "TIMESTAMP"
+                }
+            ]
+        )
+        assert result == {'ts': None}
+
+    def test_should_set_timestamp_to_none_if_invalid_value(self):
+        result = filter_record_by_schema(
+            record_object={'ts': '0000-01-01T00:00:00Z'},
```
Comment (de-code) [inline tests/unit_test/generic_web_api/transform_data_test.py]: that is the value I actually saw being logged as invalid.

File: data_pipeline/generic_web_api/transform_data.py [hunk 2]
```diff
@@ -84,6 +88,27 @@ def standardize_record_keys(record_object):
...
+def _is_valid_timestamp_string(timestamp_str: str) -> bool:
+    try:
+        if dateparser.parse(timestamp_str):
+            return True
+    except BaseException:
+        pass
+    return False
+
+
+def _get_valid_timestamp_string_or_none(
```
Comment (HazalCiplak) [inline data_pipeline/generic_web_api/transform_data.py:100]: Just wondered why do we need to have `_` beginning of this function ? And maybe in previous (`_is_valid_timestamp_string`) one too?

File: tests/unit_test/generic_web_api/transform_data_test.py (hunk 1 above)
Comment (HazalCiplak) [reply tests/unit_test/generic_web_api/transform_data_test.py]: It is a proper invalid date :)

File: tests/unit_test/generic_web_api/transform_data_test.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                    "mode": "NULLABLE",
+                    "name": "ts",
+                    "type": "TIMESTAMP"
+                }
+            ]
+        )
+        assert result == {'ts': None}
+
+    def test_should_set_timestamp_to_none_if_invalid_type(self):
+        result = filter_record_by_schema(
```
Comment (HazalCiplak) [inline tests/unit_test/generic_web_api/transform_data_test.py:37]: this test look same as `test_should_set_timestamp_to_none_if_invalid_format` or I couldn't see the difference

File: data_pipeline/generic_web_api/transform_data.py (hunk 2 above)
Comment (de-code) [reply data_pipeline/generic_web_api/transform_data.py:100]: We discussed this in our data catch-up. There are no strict rules. But my thinking was those methods, within a utility module, are there to make another function simpler. But I wouldn't want it to be used elsewhere. (Especially with the questionable use of `dateparser` which is a bit too lax)

**Comment by de-code** [comment]
I made some small improvements. Could you please re-review?

**Review by HazalCiplak — APPROVED** [review]

**Pull request merged**